Support for multiple Javadoc directories

fbenz opened the pull request:
Changes:
* Support multiple Javadoc JSON directories by supporting directories separated by `,`, e.g. `dir1,dir2,dir3`.
* Fix hard-coded use of `/`.
* Refactored `JavadocReaderImpl` for better readability.

Closes #58

jmisur commented on spring-auto-restdocs-docs/index.adoc:
```diff
@@ -44,6 +44,9 @@ Does it work with REST Assured tests?::
   Not yet. We may add support for REST Assured in the future, but your PR is also welcome.
 Is Jackson required for automatic field documentation?::
   Yes, this project only includes a Jackson visitor so far.
+Can multiple directories be configured with `javadocJsonDir`?::
```
rephrase to more generic sentence, like "Can I use auto-restdocs in multi-module projects"?
- then Explain what kind of support there is for multi-module projects.

jmisur commented on spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/javadoc/JavadocReaderImpl.java:
```diff
@@ -31,86 +33,121 @@
+    private static List<File> toAbsoluteDirs(String javadocJsonDirs) {
+        List<File> absoluteDirs = new ArrayList<>();
+        String[] dirs = split(javadocJsonDirs, PATH_DELIMITER);
+        if (dirs != null) {
```
check that input param is not blank instead of dirs []

jmisur commented on spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/javadoc/JavadocReaderImpl.java:
```diff
@@ -31,86 +33,121 @@
+        String[] dirs = split(javadocJsonDirs, PATH_DELIMITER);
+        if (dirs != null) {
+            for (String d : dirs) {
+                absoluteDirs.add(new File(d).getAbsoluteFile());
```
d -> dir
\+ trim

jmisur commented on spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/javadoc/JavadocReaderImpl.java:
```diff
@@ -31,86 +33,121 @@
 
 public class JavadocReaderImpl implements JavadocReader {
     private static final Logger log = getLogger(JavadocReader.class);
+    private static final String PATH_DELIMITER = ";";
```
for every array-like configuration property, spring always uses `comma-separated` list. check it out on https://docs.spring.io/spring-boot/docs/current-SNAPSHOT/reference/htmlsingle/

fbenz commented:
I addresses your comments. For the separator, I was just thinking about usual handling in operating systems, but it makes more sense to stick to the Spring convention here.
